### ISSUE TITLE
NX 18 build deadlocks the terminal

### ISSUE THREAD

**Issue by Aubrey-Russell** [nrwl/nx#22377]
### Current Behavior

I am trying to run nx build command and it forever deadlocks the terminal. You can't even CTRL + C it in windows, only killing it works. It says successfully completed build and then never releases the terminal handle.

![Uploading deadlock.png…]()

### Expected Behavior

It should not deadlock the terminal.

### Steps to Reproduce

1.Run nx build on a windows machine

### Nx Report

```shell
Node   : 21.1.0
OS     : win32-x64
pnpm   : 8.15.2

nx                 : 18.0.8
@nx/js             : 18.0.8
@nx/jest           : 18.0.8
@nx/linter         : 18.0.8
@nx/eslint         : 18.0.8
@nx/workspace      : 18.0.8
@nx/cypress        : 18.0.8
@nx/devkit         : 18.0.8
@nx/eslint-plugin  : 18.0.8
@nx/react          : 18.0.8
@nrwl/tao          : 18.0.8
@nx/web            : 18.0.8
@nx/webpack        : 18.0.8
typescript         : 5.4.2
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by Aubrey-Russell** [comment]
NX version update to 18.1.12 resolved the problem

**Comment by FrozenPandaz** [comment]
This was fixed by #22294 which is why it resolved after updating to `18.1.1 or 18.1.2` :wink:
